### ISSUE TITLE
0.72.5 fails to link using Xcode 14.3

### ISSUE THREAD

**Issue by colincarter** [facebook/react-native#39713]
### Description

Using Xcode 14.3 and react-native 0.72.5 the app fails to link with a `ld: library not found for -ld_classic` error.

### React Native Version

0.72.5

### Output of `npx react-native info`

```
npx react-native info
info Fetching system and libraries information...
System:
  OS: macOS 13.6
  CPU: (12) x64 Intel(R) Core(TM) i7-8850H CPU @ 2.60GHz
  Memory: 1022.44 MB / 16.00 GB
  Shell:
    version: "5.9"
    path: /bin/zsh
Binaries:
  Node:
    version: 18.16.0
    path: ~/.nvm/versions/node/v18.16.0/bin/node
  Yarn:
    version: 1.22.19
    path: /usr/local/bin/yarn
  npm:
    version: 9.5.1
    path: ~/.nvm/versions/node/v18.16.0/bin/npm
  Watchman:
    version: 2023.09.25.00
    path: /usr/local/bin/watchman
Managers:
  CocoaPods:
    version: 1.11.3
    path: /Users/colincarter/.rbenv/shims/pod
SDKs:
  iOS SDK:
    Platforms:
      - DriverKit 22.4
      - iOS 16.4
      - macOS 13.3
      - tvOS 16.4
      - watchOS 9.4
  Android SDK: Not Found
IDEs:
  Android Studio: 2022.3 AI-223.8836.35.2231.10671973
  Xcode:
    version: 14.3.1/14E300c
    path: /usr/bin/xcodebuild
Languages:
  Java:
    version: 11.0.16.1
    path: /Users/colincarter/.sdkman/candidates/java/current/bin/javac
  Ruby:
    version: 3.2.2
    path: /Users/colincarter/.rbenv/shims/ruby
npmPackages:
  "@react-native-community/cli": Not Found
  react:
    installed: 18.2.0
    wanted: 18.2.0
  react-native:
    installed: 0.72.5
    wanted: 0.72.5
  react-native-macos: Not Found
npmGlobalPackages:
  "*react-native*": Not Found
Android:
  hermesEnabled: true
  newArchEnabled: false
iOS:
  hermesEnabled: true
  newArchEnabled: false
```

### Steps to reproduce

- Create a new project with `npx react-native@latest init AwesomeProject`
- With Xcode 14.3 compile project either in Xcode or with `npx react-native run-ios`
- See linker error `ld: library not found for -ld_classic` 

### Snack, screenshot, or link to a repository

<img width="814" alt="Screenshot 2023-09-29 at 10 21 35" src="https://github.com/facebook/react-native/assets/47927/f1bd70cf-83ff-44d4-b139-59b001c8f54c">

<img width="1680" alt="Screenshot 2023-09-29 at 10 22 30" src="https://github.com/facebook/react-native/assets/47927/ddf36da8-a9b1-428c-a255-ac4d35ffe48a">

https://github.com/colincarter/AwesomeProject

**Comment by qs-wang** [comment]
I am facing same issue.

**Comment by qs-wang** [comment]
I removed the ""-ld_classic" in project file and the build was running successfully.

**Comment by cipolleschi** [comment]
This is a known issue at the moment. The fix is already in main with 63884b4 and the release crew will release a new version of RN 0.72 whenever they can.

**Comment by monochrome-yeh** [comment]
> I removed the ""-ld_classic" in project file and the build was running successfully.

Do you mind if you could tell where to remove "-ld_classic" ?
